src/main/java/com/azure/spring/reference/intellij/SpringReferenceIntellijApplication.java

```diff
@@ -262,7 +262,9 @@ private Map<String, String> buildLinks(SpringArtifactSpec springArtifactSpec, Ve
             } else {
                 LOGGER.warn("No github link set for {}", artifactId);
             }
-            if (springArtifactSpec.isHasSampleLink() && springArtifactSpec.getSampleLink() != null) {
+            if (springArtifactSpec.isHasSampleLink() && !springArtifactSpec.isBom()) {
+                result.add(new SampleLink(artifactId, version, springArtifactSpec.getAzureService()));
+            } else if (springArtifactSpec.isHasSampleLink() && springArtifactSpec.getSampleLink() != null) {
```

Comment:
The 3 added artifacts compatibilities under `spring.cloud.azure.reference.compatibilities` should be overridden against with inherited from `spring-cloud-azure-dependencies`, the new mapping is used if the artifact properties `spring.cloud.azure.reference.spring-artifacts.v4` is set to false,
https://github.com/saragluna/spring-reference-intellij/blob/5f81a2eb73682d2d7daff1f1a106dd215bbf0adc/src/main/java/com/azure/spring/reference/intellij/SpringReferenceIntellijApplication.java#L294-L296

The sample link generation will be affected, it will not be a version-tagged link, such as : https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main/keyvault/azure-securtiy-keyvault-jca -> https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_2.7.0/keyvault/azure-security-keyvault-jca/
https://github.com/saragluna/spring-reference-intellij/blob/5f81a2eb73682d2d7daff1f1a106dd215bbf0adc/src/main/java/com/azure/spring/reference/intellij/SpringReferenceIntellijApplication.java#L254-L270
So I added the logic to generate the sample link according to the sample link pattern, which will be a link with the GA version tagged.